tests/ts/query-builder-api/join-methods.ts

```diff
@@ -0,0 +1,63 @@
+import { Person } from '../fixtures/person';
+import { raw } from '../../..';
+
+(async () => {
+  await Person.query()
+    .joinRelation('pets')
+    .where('pets.species', 'dog');
+
+  await Person.query()
+    .joinRelation('pets', { alias: 'p' })
+    .where('p.species', 'dog');
+
+  await Person.query()
+    .joinRelation('[pets, parent]')
+    .where('pets.species', 'dog')
+    .where('parent.name', 'Arnold');
+
+  await Person.query()
+    .joinRelation({
+      pets: true,
+      parent: true
+    })
+    .where('pets.species', 'dog')
+    .where('parent.name', 'Arnold');
+
+  await Person.query()
+    .select('persons.id', 'parent:parent.name as grandParentName')
+    .joinRelation('[pets, parent.[pets, parent]]')
+    .where('parent:pets.species', 'dog');
+
+  await Person.query()
+    .select('persons.id', 'pr:pr.name as grandParentName')
+    .joinRelation('[pets, parent.[pets, parent]]', {
+      aliases: {
+        parent: 'pr',
+        pets: 'pt'
+      }
+    })
+    .where('pr:pt.species', 'dog');
+
+  await Person.query()
+    .select('persons.id', 'pr:pr.name as grandParentName')
+    .joinRelation('[pets as pt, parent as pr.[pets as pt, parent as pr]]')
+    .where('pr:pt.species', 'dog');
+
+  await Person.query().innerJoinRelation('pets');
+  await Person.query().outerJoinRelation('pets');
+  await Person.query().leftJoinRelation('pets');
+  await Person.query().leftOuterJoinRelation('pets');
+  await Person.query().rightJoinRelation('pets');
+  await Person.query().rightOuterJoinRelation('pets');
+  await Person.query().fullOuterJoinRelation('pets');
+  await Person.query().join(raw('pets'));
```

Comment:
Typings don't currently allow `.join('pets')` - not sure if intentional or oversight. Happy to fix if an oversight.

Reply:
They are not supposed to support that. `*joinRelation` methods take relation names. Other join methods simply call the knex equivalents and know nothing about relations.